client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/logging/LogCollector.java

```diff
@@ -44,6 +45,10 @@
      * Adds new log record to local storage.
      *
      * @param record New log record object
+     *
+     * @return bucketinfo object for added record
      */
-    void addLogRecord(Log record);
+    BucketFuture<BucketInfo> addLogRecord(Log record);
+
+    void setLogDeliveryListener(LogDeliveryListener listener);
```

Comment:
Yes, it is available though the GenericKaaClient interface.
Also I will move this method to GenericLogCollector, because it doesn't require any auto-generated information.